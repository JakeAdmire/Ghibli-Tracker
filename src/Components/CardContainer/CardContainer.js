import React from 'react';

export default function CardContainer() {
  return (
    <div>
      <h1>This is your card container</h1>
    </div>
  )
}