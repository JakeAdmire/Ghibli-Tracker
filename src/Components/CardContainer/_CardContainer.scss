.CardContainer {
  display: flex;
  flex-wrap: nowrap;
  overflow-x: auto;
  padding: 20px;
  margin-right: 20px;
  &::-webkit-scrollbar { width: 90%; }
  &::-webkit-scrollbar-track {
    border-radius: 15px;
    box-shadow: inset 0px 2px 6px 0px rgba(0, 0, 0, 0.25);
  }
  &::-webkit-scrollbar-thumb {
    border-radius: 15px;
    background: $accent-color;
    transition: all .2s ease;
    &:hover { 
      background: $accent-hover; 
      cursor: grab; 
    }
    &:active { cursor: grabbing; }
  }
}