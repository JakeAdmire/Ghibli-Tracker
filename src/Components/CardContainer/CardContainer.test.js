import React from 'react';
import { CardContainer, mapStateToProps }  from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {

  let mockFilms;
  let mockState;
  let wrapper;

  describe('CardContainer', () => {

    it('should match the snapshot', () => {

      mockFilms = [{
        "vote_count": 7068,
        "id": 129,
        "video": false,
        "vote_average": 8.5,
        "title": "Spirited Away",
        "popularity": 26.943,
        "poster_path": "/oRvMaJOmapypFUcQqpgHMZA6qL9.jpg",
        "original_language": "ja",
        "original_title": "千と千尋の神隠し",
        "genre_ids": [
            16,
            10751,
            14
        ],
        "backdrop_path": "/mnpRKVSXBX6jb56nabvmGKA0Wig.jpg",
        "adult": false,
        "overview": "A young girl, Chihiro, becomes trapped in a strange new world of spirits. When her parents undergo a mysterious transformation, she must call upon the courage she never knew she had to free her family.",
        "release_date": "2001-07-20"
      }]

      wrapper = shallow(<CardContainer films={mockFilms} />)

      expect(wrapper.debug()).toMatchSnapshot();
   })
  
  })

  describe('mapStateToProps', () => {

    it('should return an array of film objects as props', () => {

      mockFilms = [{
        "vote_count": 7068,
        "id": 129,
        "video": false,
        "vote_average": 8.5,
        "title": "Spirited Away",
        "popularity": 26.943,
        "poster_path": "/oRvMaJOmapypFUcQqpgHMZA6qL9.jpg",
        "original_language": "ja",
        "original_title": "千と千尋の神隠し",
        "genre_ids": [
            16,
            10751,
            14
        ],
        "backdrop_path": "/mnpRKVSXBX6jb56nabvmGKA0Wig.jpg",
        "adult": false,
        "overview": "A young girl, Chihiro, becomes trapped in a strange new world of spirits. When her parents undergo a mysterious transformation, she must call upon the courage she never knew she had to free her family.",
        "release_date": "2001-07-20"
      }]

      mockState = {
        films: mockFilms,
        filter: 'Show all',
        user: {}
      }

      const expected = {
        films: mockFilms
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

})