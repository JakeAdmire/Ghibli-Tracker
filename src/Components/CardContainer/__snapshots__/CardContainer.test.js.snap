// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer should match the snapshot 1`] = `
"<div className=\\"CardContainer\\">
  <Link to=\\"/129\\">
    <Card poster=\\"https://image.tmdb.org/t/p/w500/oRvMaJOmapypFUcQqpgHMZA6qL9.jpg\\" title=\\"Spirited Away\\" id={129} />
  </Link>
</div>"
`;
